File: friendly_numbers.js
friendly_numbers = (x, y)=>{
    return (s(x) == s(y) & x!=y)?"Yes":"No";
}

s = (n)=>{
    t=0;
    for(i =1;i<Math.sqrt(n);i++){
        t+=(n%i)?0:(n/i) + i;
    }
    return t;
}

File: friendly_numbers.py
def friendly_numbers(x, y):
    p = a = b = 0
    while p<x+y:
        p += 1
        if x%p<1:
            a += p
        if y%p<1:
            b += p
    return "YNeos"[a!=b or x==y::2]
